tests/test_builtin_casters.py

```diff
@@ -251,6 +251,67 @@ def test_integer_casting():
         assert "incompatible function arguments" in str(excinfo.value)
 
 
+def test_int_convert():
+    class DeepThought(object):
+        def __int__(self):
+            return 42
+
+    class ShallowThought(object):
+        pass
+
+    class FuzzyThought(object):
+        def __float__(self):
+            return 41.99999
+
+    class IndexedThought(object):
+        def __index__(self):
+            return 42
+
+    class RaisingThought(object):
+        def __index__(self):
+            raise ValueError
+
+        def __int__(self):
+            return 42
+
+    convert, noconvert = m.int_passthrough, m.int_passthrough_noconvert
+
+    def require_implicit(v):
+        pytest.raises(TypeError, noconvert, v)
+
+    def cant_convert(v):
+        pytest.raises(TypeError, convert, v)
+
+    assert convert(7) == 7
+    assert noconvert(7) == 7
+    cant_convert(3.14159)
+    assert convert(DeepThought()) == 42
+    require_implicit(DeepThought())
+    cant_convert(ShallowThought())
+    cant_convert(FuzzyThought())
+    if not env.PY2:
+        # I have no clue why __index__ is not picked up by Python 2's PyIndex_check
```

Comment:
Too bad for Python 2? I don't know what went wrong, here...

This does work in Python 2.7, though:

```pycon
Python 2.7.17 (default, Sep 30 2020, 13:38:04) 
[GCC 7.5.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> x = list(range(50))
>>> class IndexedThought(object):
...     def __index__(self):
...         return 42
... 
>>> x[IndexedThought()]
42
```